### PR TITLE
Spdx reporter fun test fixes

### PR DESCRIPTION
**Pull request by fviernau**
Fix a couple of issues in the (synthetic) test setup. These are preparation for storing scan results by provenance in `OrtResult`.

Part of #5950.

### CODE REVIEW

File: plugins/reporters/spdx/src/funTest/kotlin/SpdxDocumentReporterFunTest.kt [hunk 1]
```diff
@@ -273,13 +273,16 @@ private val ortResult = OrtResult(
...
-                provenance = ArtifactProvenance(
-                    sourceArtifact = RemoteArtifact(
-                        url = "https://some-host/first-package-sources.jar",
-                        hash = Hash.create("0000000000000000000000000000000000000000")
-                    )
+                provenance = RepositoryProvenance(
+                    vcsInfo = VcsInfo(
+                        type = VcsType.GIT,
+                        revision = "master",
+                        url = "ssh://git@github.com/path/first-package-repo.git",
```
Comment (tsteenbe) [inline plugins/reporters/spdx/src/funTest/kotlin/SpdxDocumentReporterFunTest.kt:295]: You changed the VCS URL without updating line 50 in[plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.yml and line 40 in plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.json.

**Review by tsteenbe — DISMISSED** [review]

File: plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.json [hunk 2]
```diff
@@ -84,16 +84,13 @@
...
     "homepage" : "first package's homepage URL",
     "licenseConcluded" : "NOASSERTION",
     "licenseDeclared" : "BSD-3-Clause AND (MIT OR GPL-2.0-only)",
     "licenseInfoFromFiles" : [ "Apache-2.0", "BSD-2-Clause" ],
     "name" : "first-package",
-    "packageVerificationCode" : {
-      "packageVerificationCodeValue" : "0000000000000000000000000000000000000000"
-    },
-    "summary" : "A package with all supported attributes set, with a VCS URL containing a user name, and with a scan result containing two copyright findings matched to a license finding.",
+    "summary" : "A package with all supported attributes set, with a VCS URL containing a user name, and with two scan results for the VCS containing copyright findings matched to a license finding.",
```
[the remark this answers is not in the document]
Comment (fviernau) [reply plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.json:93]: Our implementation intends to attach the `summary` to all the package representations. So, the behavior is correct.

File: plugins/reporters/spdx/src/funTest/kotlin/SpdxDocumentReporterFunTest.kt (hunk 1 above)
Comment (fviernau) [reply plugins/reporters/spdx/src/funTest/kotlin/SpdxDocumentReporterFunTest.kt:295]: In SPDX we produce 3 packages per ID:

1. binary package
2. source artifact package
3. VCS package

The lines you mention refer to 1. which uses the binary artifact URL. So, a change in these lines would be unrelated / incorrect. (I've just verified again the test is green)

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.json [hunk 3]
```diff
@@ -84,16 +84,13 @@
       "referenceType" : "purl",
       "referenceLocator" : "pkg:maven/first-package-group/first-package@0.0.1"
     } ],
-    "filesAnalyzed" : true,
+    "filesAnalyzed" : false,
```
[the remark this answers is not in the document]
Comment (fviernau) [reply plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.json:87]: No, the change is IMO right. As of this change there is no more scan result associated with the source artifact. So, there is also no package verification code. And if there is no package verification code, then the reporter sets `filesAnalyzed=false`.  

Note that there are 3 representations of the package, this one is for the source artifact (ID is "SPDXRef-Package-Maven-first-package-group-first-package-0.0.1-**source-artifact**").

File: plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.yml [hunk 4]
```diff
@@ -99,19 +99,17 @@ packages:
   - referenceCategory: "PACKAGE_MANAGER"
     referenceType: "purl"
     referenceLocator: "pkg:maven/first-package-group/first-package@0.0.1"
-  filesAnalyzed: true
+  filesAnalyzed: false
```
[the remark this answers is not in the document]
Comment (fviernau) [reply plugins/reporters/spdx/src/funTest/assets/spdx-document-reporter-expected-output.spdx.yml:102]: See my above comment. It's analog.

**Pull request merged**